marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -37,7 +38,225 @@ describe('RTL Source-to-entity map tests', () => {
+      expect(getByTestId('submitSearch-name')).toBeInTheDocument();
+      expect(inputSearchEntity).toHaveValue('craft');
+      fireEvent.click(getByTestId('submitSearch-name')); //Click on Search button to apply the filter with the desired string
+
+      //Check if the expected values are available/not available in search result.
+      expect(getByText('items')).toBeInTheDocument();
+      expect(getByText('itemTypes')).toBeInTheDocument();
+      expect(getByText('itemCategory')).toBeInTheDocument();
+      expect(getByText('Craft')).toBeInTheDocument();
+      expect(queryByText('propId')).not.toBeInTheDocument(); 
+      expect(queryByText('propName')).not.toBeInTheDocument();
+
+      //Check if the source table properties are not affected by the filter on Entity table
+      expect(getByText('proteinId')).toBeInTheDocument();
+      expect(getByText('@proteinType')).toBeInTheDocument();
+      expect(getByText('FirstNamePreferred')).toBeInTheDocument();
+      expect(getByText('LastName')).toBeInTheDocument();
+
+       //Reset the search filter on Entity table
+       fireEvent.click(entityfilterIcon);
+       let resetEntitySearch = getByTestId('ResetSearch-name');
+       expect(resetEntitySearch).toBeInTheDocument();
+       fireEvent.click(resetEntitySearch);
+ 
+       //Check if the expected values are present now after resetting the filter on Entity table.
+       expect(getByText('propId')).toBeInTheDocument();
+       expect(getByText('propName')).toBeInTheDocument();
+    });
+
+    test('Column option selector in Entity table',() => {
+      
+      const { getByText,getByTestId,queryByText } = render(<SourceToEntityMap {...data.mapProps} 
+        mappingVisible={true} 
+        entityTypeProperties={data.mapProps.entityTypePropertiesUpdated} 
+        sourceData={data.mapProps.sourceDataUpdated}/>);
+
+      //Set the data for texting in xpath expression
+
+      let exp = getByText('testNameInExp');
+      expect(exp).toBeInTheDocument();
+      fireEvent.change(exp, { target: {value: "concat(propName,'-NEW')" }});
+      fireEvent.blur(exp);
+      expect(getByText("concat(propName,'-NEW')")).toBeInTheDocument();
+      
+      /* Test column option selector in Entity table  */
+      let colOptSelect = getByText('Column Options');
+      expect(colOptSelect).toBeInTheDocument();
+      fireEvent.click(colOptSelect);
+      let Name = getByTestId('columnOptionsCheckBox-name');
+      let Type = getByTestId('columnOptionsCheckBox-type');
+      let XPathExpression = getByTestId('columnOptionsCheckBox-key');
+      let Value = getByTestId('columnOptionsCheckBox-value');
+      expect(Name).toBeInTheDocument();
+      expect(Type).toBeInTheDocument();
+      expect(XPathExpression).toBeInTheDocument();
+      expect(Value).toBeInTheDocument();
+
+      fireEvent.click(Name); //Uncheck Name column
```

Comment:
can add filtering and then uncheck to verify column doesnt exist and filter gets reset.

Reply:
Done. Added the test scenario in column option selector test.